crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,191 @@
+        _epoch: EpochId,
+        _max_epoch_upper_bound_delta: Option<u64>,
+    ) -> SuiResult {
+        Ok(())
+    }
+
+    /// Verify an intent message of a transaction with an zk login authenticator.
+    fn verify_claims<T>(
+        &self,
+        intent_msg: &IntentMessage<T>,
+        author: SuiAddress,
+        _aux_verify_data: &VerifyParams,
+        _zklogin_inputs_cache: Arc<VerifiedDigestCache<ZKLoginInputsDigest>>,
+    ) -> SuiResult
+    where
+        T: Serialize,
+    {
+        let client_data_json: CollectedClientData = serde_json::from_slice(&self.client_data)
+            .map_err(|_| SuiError::InvalidSignature {
+                error: "get client data".to_string(),
+            })?;
+        let parsed_challenge =
+            Base64UrlUnpadded::decode_vec(&client_data_json.challenge).map_err(|e| {
+                SuiError::InvalidSignature {
+                    error: e.to_string(),
+                }
+            })?;
+        let mut hasher = DefaultHash::default();
+        hasher.update(&bcs::to_bytes(&intent_msg).expect("Message serialization should not fail"));
+        let digest = hasher.finalize().digest;
+
+        // check client_data.challenge == sha256(intent_msg(tx))
+        if parsed_challenge != digest {
+            return Err(SuiError::InvalidSignature {
+                error: "invalid challenge".to_string(),
+            });
+        };
+
+        // msg = authenticator_data || sha256(client_data)
+        let mut message = self.authenticator_data.clone();
+        let client_data_hash = Sha256::digest(self.client_data.as_slice()).digest;
+        message.extend_from_slice(&client_data_hash);
+
+        match self.user_signature.scheme() {
+            SignatureScheme::Secp256r1 => {
+                let pk = Secp256r1PublicKey::from_bytes(self.user_signature.public_key_bytes())
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "Invalid r1 pk bytes".to_string(),
+                    })?;
+                if author
+                    != SuiAddress::from(
+                        &PublicKey::try_from_bytes(
+                            SignatureScheme::Secp256r1,
+                            self.user_signature.public_key_bytes(),
+                        )
+                        .unwrap(),
+                    )
+                {
+                    return Err(SuiError::InvalidSignature {
+                        error: "Invalid author".to_string(),
+                    });
+                };
+                let sig = Secp256r1Signature::from_bytes(self.user_signature.signature_bytes())
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "Invalid r1 signature bytes".to_string(),
+                    })?;
+
+                pk.verify(&message, &sig)
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "verify failed".to_string(),
+                    })
```

Comment:
Can we maybe include a detailed human-readable spec or document for how all these different pieces work together? If this already exists i'd love a pointer.

Reply:
Its a little complicate to see how this signature is indeed signing over the txn itself since this "message" doesn't actually include serializing the txn intent message but rather the client and auth data. (although it does look like its included in the client data?)

Reply:
yeah thats internal to passkey's logic so this is the reason that we cannot reuse our old r1 signature authenticator